src/fidesops/api/v1/endpoints/dataset_endpoints.py

```diff
@@ -87,9 +92,19 @@ def validate_dataset(
     try:
         # Attempt to generate a traversal for this dataset by providing an empty
         # dictionary of all unique identity keys
-        graph = DatasetGraph(convert_dataset_to_graph(dataset, connection_config.key))
-        unique_identities = set(graph.identity_keys.values())
-        Traversal(graph, {k: None for k in unique_identities})
+        graph = convert_dataset_to_graph(dataset, connection_config.key)
+
+        # Datasets for SaaS connections need to be merged with a SaaS config to
+        # be able to generate a valid traversal
+        if connection_config.connection_type == ConnectionType.saas:
+            __validate_saas_dataset(connection_config, dataset)
+            graph = merge_datasets(
+                graph,
+                connection_config.get_saas_config().get_graph(),
+            )
```

Comment:
For future consideration, I still think it would be nice to see a representation of the merged dataset somewhere in an API response.  If there are issues, being able to see how we merged the dataset might be useful.  However, I understand that the merged dataset is of type Dataset, not FidesopsDataset, making it difficult to return the merge here, so this could be a follow-up.

This is a positive step forward though, merging the Datasets before building the traversal for saas configs.